Feature/mapping test

foxleigh81 opened the pull request:
## 🚀 Overview: 
Puts the `Mapping` component from the `components` package into the `client` pacakge.

## 🤔 Reason: 
To make sure everything works as expected.

## Link to preview:
https://serge-review-feature-ma-t1rwsv.herokuapp.com/?wargame=wargame-kaf9gvb3&access=p5543

## 🔨Work carried out:

- [x] Updates the `config` and `helpers` packages to build to js so they can be used as part of the wider `serge` package ecosystem without relying on TS.
- [x] Updates styling on the `mapping` component to be less storybook specific.
- [x] Adds a decorator to the `mapping` component stories to restore the required styling for storybook
- [x] Imports the `mapping` component into the `client` package and test ran it.
- [ ] Changing the perceived force of an asset is reflected on the map 
- [x] Tests pass

## 🖥️ Screenshot
![Screenshot 2020-05-19 at 14 01 54](https://user-images.githubusercontent.com/151028/82329485-4dc60f00-99d9-11ea-8415-6b74f09ed81e.png)

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have assigned myself to this PR.
- [x] I have chosen an appropriate label for the PR.
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.
- [x] I confirm that I have checked for required README updates and acted accordingly.

## 📝 Developer Notes:
- This just imports the blank map, however it works as expected with the existing data. Importing the child components will probably be where the data mismatch becomes an issue.

IanMayo commented:
@foxleigh81 - I've added a test to the description. It uses `change perception` to verify the server round-trip.   We can send this message without all the _plan route_ functionality that I still have to implement.

We can trigger the "round-trip" to send the perceived update to the backend by reproducing the `perceivedStateCallback` method in `client\components\mapping\index.jsx`.  The perceived form just needs to re-package the `PerceivedDataValues` into the structure used in `perceivedStateCallback`, and fire it into a `sendMessage()` method.

Aah, there's a point. The `mapping` component could do with the `sendMessage` functionality that's in the legacy mapping component. It wraps the message in a uniform container, then sends it to the backend processing. 

So, I think the new mapping component should no longer pass the `postback` directly to the context.  It should wrap that function in the above `sendMessage` function, and pass the `sendMessage` to the mapping context.

foxleigh81 commented:
> @foxleigh81 - I've added a test to the description. It uses `change perception` to verify the server round-trip. We can send this message without all the _plan route_ functionality that I still have to implement.
> 
> We can trigger the "round-trip" to send the perceived update to the backend by reproducing the `perceivedStateCallback` method in `client\components\mapping\index.jsx`. The perceived form just needs to re-package the `PerceivedDataValues` into the structure used in `perceivedStateCallback`, and fire it into a `sendMessage()` method.
> 
> Aah, there's a point. The `mapping` component could do with the `sendMessage` functionality that's in the legacy mapping component. It wraps the message in a uniform container, then sends it to the backend processing.
> 
> So, I think the new mapping component should no longer pass the `postback` directly to the context. It should wrap that function in the above `sendMessage` function, and pass the `sendMessage` to the mapping context.

Cool. You put it in the wrong place, but I've moved it :)

IanMayo commented:
Great, this is really exciting.

The data in the perception form isn't editable.  I've added a 'preview' link to the issue description that goes straight in as a Blue player - so it's quick & easy to get to a point where perception can be tested.

IanMayo committed:
Introduce 'hosting' element, plus new platform-type for UAS

IanMayo committed:
introduce task-group object

IanMayo committed:
introduce task group object

IanMayo committed:
remove stray char

IanMayo committed:
correct platform type

IanMayo committed:
Reflect new forces mock

IanMayo committed:
Update tests to match new mock data

IanMayo committed:
introduce (Temporary) task-group icon

IanMayo committed:
introduce icon for task group

IanMayo committed:
introduce more wargame content

IanMayo committed:
use correct platform type

IanMayo committed:
use correct platform type

IanMayo committed:
Introduce route-child element, display them in World State

IanMayo committed:
Give red force a hosted platform

IanMayo committed:
Give red force a hosted platform

IanMayo committed:
recognise increased number of platform types

IanMayo committed:
recognise increased number of platform types

IanMayo committed:
linter improvements

IanMayo committed:
introduce package command to run watch for tests

IanMayo committed:
Give platforms unique perceived names

IanMayo committed:
minor formatting improvement

IanMayo committed:
introduce task group marker

IanMayo committed:
refactor out utility function

IanMayo committed:
valid handling of hosted assets

IanMayo committed:
Support for comprising in data

IanMayo committed:
For hosted & comprised assets, we need to provide the location of the parent

IanMayo committed:
Decide which assets to show based upon the RouteStore - since it usefully does some pre-processing

IanMayo committed:
linter tidying

IanMayo committed:
Introduce valid storybook for WorldState form

IanMayo committed:
Improve display of child assets

IanMayo committed:
Let a comprising platform host an asset

IanMayo committed:
Check that other force can see assets hosted on assets they perceive

IanMayo committed:
rename uas, to match working practice

IanMayo committed:
make UAS icon file name match the platform-type

IanMayo committed:
Improve presentation, and introduce more complex data

IanMayo committed:
linting

IanMayo committed:
Introduce handler for view-as

IanMayo committed:
Only respect viewAs when white force

IanMayo committed:
don't use light/dark themes, but replace umpire color with something darker when necessary

IanMayo committed:
change strategy. Don't use routes.  Instead, do perception based on forces data.  This change was to prevent the "routes" data being cleared when the ViewAs value changed.

IanMayo committed:
don't recalculate route-data on view-as changes

IanMayo committed:
make map taller

IanMayo committed:
allow route color to be overwritten, for when player is in View-As mode

IanMayo committed:
Independently collate a mock set of routes that can be modified to reflect View-As setting

IanMayo committed:
linter fixes

IanMayo committed:
remove debug lines

IanMayo committed:
Display selected asset, streamline View As logic

IanMayo committed:
Mark current selectedAs button as selected

IanMayo committed:
Introduce toggle for if current player is umpire role

lilitkarapetyan committed:
fix child items

lilitkarapetyan committed:
fix click event

lilitkarapetyan committed:
collapsible

lilitkarapetyan committed:
collapsible

lilitkarapetyan committed:
collapsible

lilitkarapetyan committed:
fix warnings

IanMayo committed:
Remove labels

IanMayo committed:
Store snaps

IanMayo committed:
Make assets in world state visible

IanMayo committed:
Introduce new initialised version of Task Group wargame

lilitkarapetyan committed:
add display name for components

lilitkarapetyan committed:
autofixes

IanMayo committed:
also store `hosting` for child platforms

IanMayo committed:
also store hosting for child items

IanMayo committed:
Introduce task-force member that is hosting other assets.
Introduce initialised version of new wargame,

IanMayo committed:
update snap

IanMayo committed:
Merge pull request #452 from serge-web/feature/450_task_group_data

Feature/450 task group data

IanMayo committed:
pass userIsUmpire to collatePerceivedData

IanMayo committed:
Perceived form should show perceived type/force/name for asset, not real one

IanMayo committed:
linter fixes

IanMayo committed:
Verify handling for umpire

IanMayo committed:
fix linter

IanMayo committed:
world state panel outer border to tab color

IanMayo committed:
Make planning/perception forms have consistent styling

IanMayo committed:
put white box around color shades, to improve contrast

IanMayo committed:
Merge pull request #463 from serge-web/hotfix/462_UI_fixes

Hotfix/462 ui fixes

IanMayo committed:
Merge pull request #461 from serge-web/fix/460_not_perceived_name

Fix/460 not perceived name

lilitkarapetyan committed:
save

lilitkarapetyan committed:
drag & drop

lilitkarapetyan committed:
group component

lilitkarapetyan committed:
move in to group

IanMayo committed:
publicise get platform type for

IanMayo committed:
Planned routes should be to achievable destinations

IanMayo committed:
Planned routes should be to achievable destinations

IanMayo committed:
use correct values for new leg

IanMayo committed:
don't display a speed label if there isn't a speed

IanMayo committed:
Restructure, to allow child element to be referenced

IanMayo committed:
Support planning legs for asset with unlimited travel distance

IanMayo committed:
apply linter

IanMayo committed:
linter fixes

IanMayo committed:
Experiment iwth alternate definition

IanMayo committed:
provide remaining return types

IanMayo committed:
linter fixes

IanMayo committed:
trial alternate way of expressing return type

IanMayo committed:
Merge pull request #466 from serge-web/feature/458_plan_aircraft_routes

Feature/458 plan aircraft routes

IanMayo committed:
refactor, to reduce rendering

IanMayo committed:
linter fixes

IanMayo committed:
Merge pull request #468 from serge-web/hotfix/expensive_renders

Hotfix/expensive renders

IanMayo committed:
Hide planning form while planning is conducted

IanMayo committed:
fix linting

IanMayo committed:
Correctly collate data for test

IanMayo committed:
reduce height of map, so plan turn form is in visible region

lilitkarapetyan committed:
Groups component

lilitkarapetyan committed:
remove from group logick

lilitkarapetyan committed:
multilevel groups

IanMayo committed:
use lower case force comparison

IanMayo committed:
clear selection when map is clicked on

IanMayo committed:
avoid situation where state update meant we were trying to display the wrong form

IanMayo committed:
Linter fixes

IanMayo committed:
clear planning artifacts if no asset selected

IanMayo committed:
only show shadow for selected asset

IanMayo committed:
linter fixes

IanMayo committed:
Provide speeds for Task Group statuses

IanMayo committed:
fix for planning part of turn

IanMayo committed:
linter tidying

IanMayo committed:
Initial logic for enhanced turn ends

IanMayo committed:
Initial logic for enhanced turn ends

IanMayo committed:
introduce new data type, to ease task of calculating turn markers

IanMayo committed:
refined implementation

IanMayo committed:
change how data is collated for turn markers

IanMayo committed:
trim the first cell of the route, it's the starting cell

IanMayo committed:
linter fixes

IanMayo committed:
plan routes using asset color

IanMayo committed:
remove fills, so they're asset-dependent

IanMayo committed:
darken the shade for the planning route

IanMayo committed:
linter fixes

IanMayo committed:
Use transparency to generate darker shades in route planning

IanMayo committed:
Merge pull request #473 from serge-web/feature/471_allowable_cell_color

Feature/471 allowable cell color

IanMayo committed:
Merge pull request #469 from serge-web/feature/467_hide_plan_turn_form

Feature/467 hide plan turn form

lilitkarapetyan committed:
fix firefox mouse position

lilitkarapetyan committed:
planed turns and status

lilitkarapetyan committed:
str replace issue fix

IanMayo committed:
minor comment tidying, to drive update

IanMayo committed:
CSS Styling

IanMayo committed:
remove operator that linter didn't like

IanMayo committed:
More linter fixes

IanMayo committed:
Try to fix linter issues

IanMayo committed:
More linter fixes

IanMayo committed:
Correct return type

IanMayo committed:
More linter fixes

IanMayo committed:
linter fixes

IanMayo committed:
Catching missing type-renaming

IanMayo committed:
Fix remaining type-rename

IanMayo committed:
Try to fix conflict

IanMayo committed:
Rename types, to reduce confusion

lilitkarapetyan committed:
forceName update on group_out

lilitkarapetyan committed:
fix group forceName key

lilitkarapetyan committed:
add force name on group event

lilitkarapetyan committed:
platformType for new group

lilitkarapetyan committed:
autocollapse on dragHover

lilitkarapetyan committed:
add showOtherPlatforms switcher

IanMayo committed:
Merge pull request #474 from serge-web/feature/459-task-group-interactions

Feature/459 task group interactions

IanMayo committed:
hotfix - don't take color if no selected asset

IanMayo committed:
fixes for linter

IanMayo committed:
Fix linter errors

IanMayo committed:
linter fixes

IanMayo committed:
hotfix applied

IanMayo committed:
Merge branch 'hotfix_linter_for_scss' into hotfix_for_planning_airborne_route

IanMayo committed:
auto-generated file

IanMayo committed:
Merge pull request #480 from serge-web/hotfix_for_planning_airborne_route

Hotfix for planning airborne route

IanMayo committed:
Improve turn numbering for planned routes (esp track with no planned steps)

IanMayo committed:
linter tidying

IanMayo committed:
Merge pull request #481 from serge-web/hotfix_for_planning_airborne_route

Hotfix for planning airborne route

IanMayo committed:
When route is cleared, we also need to clear allowable filtered cells

IanMayo committed:
Provide updated planning constraints when route cleared

IanMayo committed:
linter fixes

IanMayo committed:
improve variable naming

IanMayo committed:
early steps towards embedding location in the RouteStore

IanMayo committed:
partially working. Commit before running yarn upgrade

IanMayo committed:
linter fixes

IanMayo committed:
linter fixes

IanMayo committed:
linter fixes

IanMayo committed:
remove debug line

IanMayo committed:
Remove debug lines, fix tests

IanMayo committed:
Update node dependencies

IanMayo committed:
Update snap tests

IanMayo committed:
Make children optional

IanMayo committed:
Merge pull request #483 from serge-web/hotfix/conditional_wrapper

Hotfix - new TypeScript checking failure

IanMayo committed:
Merge branch 'feature/mapping-test' into 441_declutter

lilitkarapetyan committed:
fix props

lilitkarapetyan committed:
fix es version and jest version for node 12.x

IanMayo committed:
More implementation

IanMayo committed:
Merge branch 'feature/mapping-test' into feature/mapping-test-fix-props

# Conflicts:
#	packages/components/src/local/form-elements/rcb/types/conditional-wrapper.d.ts
#	yarn.lock

IanMayo committed:
Fix snap

IanMayo committed:
Merge pull request #484 from serge-web/feature/mapping-test-fix-props

Feature/mapping test fix props

IanMayo committed:
Merge branch 'feature/mapping-test' into 441_declutter

IanMayo committed:
drop debug line

IanMayo committed:
Pass tilesize to declutter algorithm

IanMayo committed:
Improve decluttering

IanMayo committed:
tidying

IanMayo committed:
Switch to positions

IanMayo committed:
Switch to positions

IanMayo committed:
Reflect move to positions over hex centres

IanMayo committed:
use correct getter for hex from grid

IanMayo committed:
remove debug statements

IanMayo committed:
Remove debug lines

IanMayo committed:
we no longer leave the first one at the centre

IanMayo committed:
If just two points in cluster, separate them horizontally instead of vertically

IanMayo committed:
Introduce flag for whether to show planned routes. Don't show them for assets of other forces.

IanMayo committed:
Merge pull request #485 from serge-web/441_declutter

441 declutter

IanMayo committed:
Early steps - get UI working

IanMayo committed:
Merge branch 'feature/mapping-test' into feature/487_filter_planned_steps

IanMayo committed:
make button color little lighter, so it shows in reverse text

IanMayo committed:
linter tidying

IanMayo committed:
include tests to verify filtering planned routes

IanMayo committed:
update tests to match new UI

IanMayo committed:
use new state for filter planned routes

IanMayo committed:
refactor to allow a single turn to be handled, or all turns

IanMayo committed:
updates for new UI

IanMayo committed:
pass filter planned steps parameter

IanMayo committed:
Give some history to task group - to check filtering

IanMayo committed:
Meet new API requirements for create store

IanMayo committed:
Just do a single step when filtering is on

IanMayo committed:
Introduce support for filtered history

IanMayo committed:
reflect new extended history for Dhow-A

IanMayo committed:
Give some more extended history for red platform

IanMayo committed:
listen out for history flag changing

IanMayo committed:
Fixes

IanMayo committed:
Minor tidying

IanMayo committed:
remove special handling of displayed steps in adjudication phases

IanMayo committed:
Introduce wider range of formats for history data

IanMayo committed:
introduce instance of other way of representing history

IanMayo committed:
Improve variable naming

IanMayo committed:
introduce testing of routes for

IanMayo committed:
Put history in correct order

IanMayo committed:
Change dummy locations, so tests can determine correct one in use

IanMayo committed:
Improve history location

IanMayo committed:
Refactor data storage types

IanMayo committed:
Refactor data storage types

IanMayo committed:
Improve logic in how we display history steps

IanMayo committed:
Refactor into two different methods for planned & historic routes

IanMayo committed:
Refactor, to avoid problem where jest not finding tests

IanMayo committed:
linter tidying

IanMayo committed:
linter fixes

IanMayo committed:
API fixes

IanMayo committed:
remove unrecognised method

IanMayo committed:
Merge pull request #488 from serge-web/feature/487_filter_planned_steps

Feature/487 filter planned steps

IanMayo committed:
Modify create route so that it's aware of selected asset, so it can return all planned steps for selected asset

IanMayo committed:
Introduce tests for displaying all planned steps for selected asset

IanMayo committed:
Reflect updated API

IanMayo committed:
Don't declutter the last point of the last step of the selected track - since we want the planning marker to appear on the end of the track line

IanMayo committed:
Don't forget to declutter history tracks

IanMayo committed:
improve TypeScript

IanMayo committed:
Remove MPA history, so it doesn't get decluttered

IanMayo committed:
minor tidying

IanMayo committed:
Merge pull request #489 from serge-web/feature/486_declutter_planning_marker

Feature/486 declutter planning marker

IanMayo committed:
update forces & platform types

IanMayo committed:
Introduce initialised version of Task Group wargame

IanMayo committed:
Add new wargame

IanMayo committed:
Update forces & platform types

IanMayo committed:
Merge pull request #491 from serge-web/hotfix/update_scenario

Hotfix/update scenario
